packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -215,6 +217,25 @@ class Analytics implements IAnalytics {
     this.clientDataStore = this.storeManager.getStore('clientData') as Store;
     this.userSessionManager.init(this.clientDataStore);
 
+    // Get user selected consent manager
+    const selectedConsentManager: string | undefined = getUserSelectedConsentManager(
+      state.consents.cookieConsentOptions.value,
+    );
+
+    if (selectedConsentManager) {
+      if (!SUPPORTED_CONSENT_MANAGERS.includes(selectedConsentManager)) {
+        this.logger?.error(`[ConsentManager]:: Provided consent manager is not supported.`);
+      } else {
+        this.pluginsManager.invokeSingle(
+          `consentManager.init`,
+          state,
+          this.pluginsManager,
+          selectedConsentManager,
+          this.logger,
+        );
+      }
+    }
+
```

Comment:
The plugin manager already loads the selected consent manager plugin. There is no need to deduce the actual consent manager again here.

Here, we should simply invoke the consent manager plugin.

Reply:
We need to deduce it here to pass it down to the consent manager as an argument.

Reply:
Yeah, that's not required, is what I meant.